persistence/provider/boltdb/transaction.go

```diff
@@ -11,27 +11,33 @@ import (
 // data stores.
 type transaction struct {
 	ds     *dataStore
+	result *persistence.TransactionResult
 	appKey []byte
 	actual *bbolt.Tx
 }
 
 // Commit applies the changes from the transaction.
-func (t *transaction) Commit(ctx context.Context) error {
+func (t *transaction) Commit(
+	ctx context.Context,
+) (*persistence.TransactionResult, error) {
 	defer t.end()
 
 	if t.ds == nil {
-		return persistence.ErrTransactionClosed
+		return nil, persistence.ErrTransactionClosed
 	}
 
 	if err := t.ds.checkOpen(); err != nil {
-		return err
+		return nil, err
 	}
 
 	if t.actual != nil {
-		return t.actual.Commit()
+		if err := t.actual.Commit(); err != nil {
+			return nil, err
+		}
+
```

Comment:
This would get rid of those those uncoverable lines. I wouldn't recommend that if it was something we could reproduce in any sane way or we were responsible for generating the error in our code.

```suggestion
		return &t.result, t.actual.Commit()
```